client/app/domain/options/delete/domain-options-delete.controller.js

```diff
@@ -0,0 +1,44 @@
+angular.module('controllers').controller(
+  'controllers.Domain.Options.Delete',
+  class DomainDnsLockCtrl {
+    constructor($scope, $rootScope, $translate, Alerter, Domain) {
+      this.$scope = $scope;
+      this.$rootScope = $rootScope;
+      this.$translate = $translate;
+      this.Alerter = Alerter;
+      this.Domain = Domain;
+    }
+
+    $onInit() {
+      this.loading = false;
+      this.domain = this.$scope.currentActionData.domain;
+      this.option = this.$scope.currentActionData.option;
+      this.$scope.deleteDomain = () => this.deleteDomain();
+    }
+
+    deleteDomain() {
+      this.loading = true;
+      return this.Domain.deleteOption(
+        this.domain.name,
+        this.option.option,
+      )
+        .then(() => this.Alerter.success(
+          this.$translate.instant('domain_tab_options_delete_success'),
+          this.$scope.alerts.main,
+        ))
+        .catch((err) => {
+          _.set(err, 'type', err.type || 'ERROR');
```

Comment:
You should avoid updating `err`, which is an input (cf [PURE functions](https://www.nicoespeon.com/en/2015/01/pure-functions-javascript/))

For example, you can do something like that instead:

```javascript
this.Alerter.alertFromSWS(
	this.$translate.instant('domain_tab_options_delete_error'),
	Object.assign({}, err, { type: err.type || 'ERROR' }),
	this.$scope.alerts.main,
);
```

Reply:
Thanks for that. This change is done